sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -723,6 +725,11 @@ class SparkSession private(
    * @since 2.0.0
    */
   def stop(): Unit = {
+    if (!stopped.compareAndSet(false, true)) {
+      logInfo("SparkSession already stopped.")
```

Comment:
Actually, this function is about `Stop the underlying SparkContext`. And, `SparkContext.stop` already have `     logInfo("SparkContext already stopped.")`, too. Is this message for the case with `SparkSession. stopped` is true and `SparkContext.stopped`?

Reply:
Idea was to prevent multiple invocations of SparkSession.stop to prevent multiple concurrent invocations of `stopStreamingQueries`. This prints a session specific message if multiple stop is invoked on the session.

Reply:
To avoid it I would need to introduce a new hook method that `SparkSession` registers with `SparkContext` which is invoked when `SparkContext.stop` is invoked. I think that would also handle a few corner cases. Can I explore that option?

Reply:
I refactored a bit and remove these checks completely. It should be handled inside `StreamingQueryManager.stopActiveQueries`